src/app/components/pages/AdminUserManager.tsx

```diff
@@ -110,18 +235,41 @@ export const AdminUserManager = () => {
   const modifyUserEmailVerificationStatusesAndUpdateResults = async (status: EmailVerificationStatus) => {
     setUserUpdating(true);
     const selectedEmails =
-      searchResults?.filter((user) => user.id && selectedUserIds.includes(user.id)).map((user) => user.email || "") ||
+      searchResults?.filter((user) => user.id && selectedUserIds.includes(user.id)).map((user) => user.email ?? "") ??
       [];
     await dispatch(adminModifyUserEmailVerificationStatuses(status, selectedEmails));
     dispatch(adminUserSearchRequest(searchQuery));
     setSelectedUserIds([]);
     setUserUpdating(false);
   };
 
-  const search = (event: React.FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    setSearchRequested(true);
+  const declineTeacherUpgradeAndUpdateResults = async () => {
+    setUserUpdating(true);
+    const response = await dispatch(adminModifyTeacherPending(false, selectedUserIds));
     dispatch(adminUserSearchRequest(searchQuery));
+    setSelectedUserIds([]);
+    setUserUpdating(false);
+    if (response) dispatch(showSuccessToast("Teacher Pending Status", response));
+  };
+
+  const confirmUnverifiedUserPromotions = function () {
+    if (searchResults.length) {
+      const unverifiedSelectedUsers = selectedUserIds
+        .map((selectedId) => searchResults.filter((result) => result.id === selectedId)[0])
```

Comment:
The problem with `find` is the potential for undefined results, which `filter` didn't have.  I've rewritten this however using `flatmap` and `find` which I think should be more efficient since it handles the undefined options and also avoid iterating over the array multiple times

Reply:
> The problem with find is the potential for undefined results, which filter didn't have.

The problem is the same in both: if `find` returns `undefined`, it means that it couldn't find anything. If `filter` returns an empty array, it means that it couldn't find anything, and it will fail when you try to access the first element (`[0]`). Given that you weren't checking for that possibility before, I'll assume that it's not a valid scenario. So in that case, `find` will never return `undefined` either.

`flatmap` and `find` is more efficient, but it's also a bit hacky. Since in this case efficiency is not our top priority, I would opt for writing something that's more readable. For instance, this should be equivalent, and it's easier to understand what it's doing:

```typescript
const unverifiedSelectedUsers = selectedUserIds
  .map((selectedId) => searchResults.find((user) => user.id === selectedId))
  .filter((user): user is UserSummaryForAdminUsersDTO => user?.emailVerificationStatus !== "VERIFIED");
```

I have used a [Type Guard](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) to indicate that `filter` will always return a `UserSummaryForAdminUsersDTO`, rather than `UserSummaryForAdminUsersDTO|undefined`.

Having said all that, feel free to ignore this if you find `flatMap` + `find` more readable. It could just be me that finds it slightly hacky.

Reply:
Also, I haven't tested that, it may be that instead of `user?.emailVerificationStatus !== "VERIFIED"` you need to change it to `user !== undefined && user.emailVerificationStatus !== "VERIFIED"`, so please test this if you decide to go down this route.

Reply:
On second thought, if you know that the result of `find` will never be `undefined`, then you should add the Type Guard there, instead of in the `filter`.

Reply:
Updated to 

```
const unverifiedSelectedUsers = selectedUserIds
  .map((selectedId) => searchResults.find((user) => user.id === selectedId)!)
  .filter((user) => user.emailVerificationStatus !== "VERIFIED");
  ```
  
  Since the button to trigger this function can't be used until `searchResults` exist, and the `selectedUserIds` are selected from those search results.